# Workshop ASHG 2019, Houston, Texas



## Google Slides
[MIS_workshop](https://docs.google.com/presentation/d/1vOeT1t0_INSj9NN0RJ2EIEZbfmjLFTfXQLydDR-BTmU/edit?usp=sharing)

## Slides version 1 
[MIS_workshop_v1.pdf](https://github.com/lukfor/imputationserver-ashg19/raw/master/files/MIS_workshop_v1.pdf)

## ASHG Testserver
1) Connect to our [ASHG Testcluster](https://imputationserver.sph.umich.edu/aws) on Amazon AWS

2) Register a new username

**Note:** Your username will be active right after registration (no verification link will be sent). 

3) Download test data: [Chr20 Phased](https://github.com/genepi/imputationserver/raw/master/test-data/data/chr20-phased/chr20.R50.merged.1.330k.recode.small.vcf.gz) or [Chr20 Unphased](https://github.com/genepi/imputationserver/raw/master/test-data/data/chr20-unphased/chr20.R50.merged.1.330k.recode.unphased.small.vcf.gz)

4) Run a job using HapMap2

## Update: Oct 10, 2019

- For the interactive polls, we recommend to install the mobile [Poll Everywhere app](https://www.polleverywhere.com/mobile)
- No other additional software is needed
- For the question/answer session, we invite participants to send questions to  [christian.fuchsberger@eurac.edu](mailto:christian.fuchsberger@eurac.edu)
- Recommend to check the website the day before the workshop for last minute updates

## Date/time of workshop
Thursday, October 17, 7:15-8:45 am

## Factsheet

### Workshop facilitator(s)
- Christian Fuchsberger, [christian.fuchsberger@eurac.edu](mailto:christian.fuchsberger@eurac.edu)
- Lukas Forer, [lukas.forer@i-med.ac.at](mailto:lukas.forer@i-med.ac.at)
- Sebastian Schoenherr, [sebastian.schoenherr@i-med.ac.at](mailto:sebastian.schoenherr@i-med.ac.at)
- Cassie Spracklen, [cspracklen@umass.edu](mailto:cspracklen@umass.edu)

### Workshop description (from the program)
Genotype imputation is a key component of modern genetic studies. This interactive workshop is intended for anyone interested in learning how to use the Michigan Imputation Server (MIS; [https://imputationserver.sph.umich.edu](https://imputationserver.sph.umich.edu)) to impute genotypes and how to use the imputed genotypes, with a special focus on up-coming reference panels, including the multi-ancestry panel from the TOPMed program. A brief overview of imputation and the server will be followed by demonstrations and exercises, including:

1. quality control and preparation of genetic data for use on the MIS with a special focus on diverse ancestries and chromosome X
2. tracking runs and use of the application program interface for larger jobs
3. downloading data from the MIS and preparing data for genetic analysis
4. performing a GWAS using imputed data and interpreting results, taking into account imputation quality

We encourage participants to ask specific questions about their own projects. Workshop materials, including slides and example data sets, will be made available before the workshop and will remain online at the MIS website. We expect that this workshop will enable participants to generate high-quality imputed data sets and to properly analyze them. Attendees are expected to follow materials on their personal laptops. 
 
### Intended Audience
Attendees interested in learning how to perform genotype imputation and use imputed genotypes in their research, especially trainees. Prior experience with the analysis of genome-wide association studies (GWAS) and the unix command line are recommended for the workshop.
 
### Requirements
Device(s) that attendees should bring to the workshop (e.g., laptop, smartphone, etc.): Laptop
 
### Recommended background reading
Will be listed on this website ([https://imputationserver.sph.umich.edu/ashg19](https://imputationserver.sph.umich.edu/ashg19))

### Other information:
Recommended to check this website ([https://imputationserver.sph.umich.edu/ashg19](https://imputationserver.sph.umich.edu/ashg19)) one week before the workshop for updates.
